packages/api/src/users/commands/handlers/save-survey.handler.ts

```diff
@@ -0,0 +1,23 @@
+import { BadRequestException, NotFoundException } from '@nestjs/common';
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+
+import { User } from '@coderscamp/shared/models/user';
+
+import { ErrorMessage } from '../../error';
+import { UsersRepository } from '../../users.repository';
+import { SaveSurveyCommand } from '../save-survey.command';
+
+@CommandHandler(SaveSurveyCommand)
+export class SaveSurveyHandler implements ICommandHandler<SaveSurveyCommand> {
+  constructor(private repository: UsersRepository) {}
+
+  async execute({ input }: SaveSurveyCommand): Promise<User> {
+    const userFromDb = await this.repository.getUser(input.id);
+
+    if (!userFromDb) throw new NotFoundException(ErrorMessage.USER_NOT_FOUND);
```

Comment:
I think is not an issue to have survey for user which not exist.
More problematic is that we need to load a whole user entity just to save survey. 
If we follow this approach in the future, we will end up with loading all users related things (projects / comments) etc. in every case (lazy loading is even more evil - difficult to keep invariants / business rules).

I'd recommend removing check is user exist. 
And the next one about SURVEY_ALREADY_FILLED hmmm...
There is a short period of time between line 15 (load)  and 19, so while loading from db survey may not be filled, but in 19 yes, and the code will not detect that.  
We should decide what to do: 
- use optimistic locking
- override the survey